modules/fx/arrow-fx/src/main/kotlin/arrow/fx/extensions/io.kt

```diff
@@ -376,15 +376,15 @@ fun <A> IO.Companion.fx(c: suspend ConcurrentSyntax<IOPartialOf<Nothing>>.() ->
  * converts this Either to an IO. The resulting IO will evaluate to this Eithers
  * Right value or alternatively to the result of applying the specified function to this Left value.
  */
-fun <E, A> Either<E, A>.toIO(f: (E) -> Throwable): IO<Nothing, A> =
+fun <E, A> Either<E, A>.toIOException(f: (E) -> Throwable): IO<Nothing, A> =
```

Comment:
I think in most cases where you'd want this function `E ~ Throwable` so I'd skip the function. You can always do `mapLeft {}.toIOException` anyway.

Reply:
^^^

From the chat, this is the signature: `fun <A> Either<Throwable, A>.toIO(): IO<Nothing, A>`. Let's not add an specialization.

Reply:
Fixed ;)
I don't know why, but now test of `toIO` throw:
```
Arrow-kt internal error. Please let us know and create a ticket at https://github.com/arrow-kt/arrow/issues/new/choose
arrow.fx.IOTimedResultedException: Arrow-kt internal error. Please let us know and create a ticket at https://github.com/arrow-kt/arrow/issues/new/choose
```

Reply:
So two things. The test will except now. Use `attempt` for it instead.

The second is...that shouldn't be the error thrown by the test.

Reply:
Okay, found why. cc @nomisRev and @aballano 

In IORunLoop, when an error is found and there's no recovery mechanism, it returns the same element here: https://github.com/arrow-kt/arrow/blob/0.11.0/bio/modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IORunLoop.kt#L57

Then in `IO#unsafeRunTimedTotal()` the implementation for `RaiseError` returns `None`: https://github.com/arrow-kt/arrow/blob/0.11.0/bio/modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt#L788

Which gets folded here into an exception: https://github.com/arrow-kt/arrow/blob/0.11.0/bio/modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt#L751

Fix, make that return Either as it should: https://github.com/arrow-kt/arrow/blob/0.11.0/bio/modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt#L788

Lemme make a quick PR, see what happens :D

Reply:
https://github.com/arrow-kt/arrow/pull/2004

You can merge after that goes in!

Reply:
After applying that changes, everything went well, thanks ;)

Reply:
@pakoito I'm now a little bit confused. Most new changes are to release/0.11.0 branch, but you merged your PR to 0.11.0/bio branch. What to rebase onto to check if tests work?

Reply:
I made a mistake, I'm fixing it just now: https://github.com/arrow-kt/arrow/pull/2009